### PR TITLE
Feature/email pattern validator and bugfix on contact modal

### PR DESCRIPTION
**Pull request by emina-str**
- add customPatternValidator which excepts regex as parameter
- add min and maxLength attribute to textarea
- add emailInputValidator
- add distinctUntilChanged() to checkbox observables to not trigger them onSubmit action

### CODE REVIEW

File: alv-portal-ui/src/app/shared/forms/validation.service.ts [hunk 1]
```diff
@@ -37,7 +37,18 @@ export class ValidationService {
...
       requiredAfter: 'portal.forms.validation.ngbDate.after'
     },
-    // TODO create a emailValidator global message
+    {
+      error: 'defaultPatternValidator',
+      message: 'home.tools.job-publication.messages.validate.default-pattern-format'
+    },
+    {
+      error: 'houseNumValidator',
+      message: 'home.tools.job-publication.messages.validate.default-pattern-format'
```
[the remark this answers is not in the document]
Comment (pavax) [reply alv-portal-ui/src/app/shared/forms/validation.service.ts]: new keys should be added using a new key pattern for example
`portal,portal.general.form.validation.message.houseNumber`

File: alv-portal-ui/src/app/shared/forms/validation.service.ts [hunk 2]
```diff
@@ -37,7 +37,18 @@ export class ValidationService {
...
+    },
+    {
+      error: 'emailValidator',
+      message: 'home.tools.job-publication.messages.validate.email-format'
```
Comment (99999989) [inline alv-portal-ui/src/app/shared/forms/validation.service.ts]: Use the global error message `global.messages.validate.email.invalid`

File: alv-portal-ui/src/app/shared/forms/validation.service.ts [hunk 3]
```diff
@@ -37,7 +37,18 @@ export class ValidationService {
...
       requiredAfter: 'portal.forms.validation.ngbDate.after'
     },
-    // TODO create a emailValidator global message
+    {
+      error: 'defaultPatternValidator',
+      message: 'home.tools.job-publication.messages.validate.default-pattern-format'
+    },
+    {
+      error: 'houseNumValidator',
+      message: 'home.tools.job-publication.messages.validate.housenr'
```
Comment (99999989) [inline alv-portal-ui/src/app/shared/forms/validation.service.ts]: Create a global error message in the `portal` namespace: e.g. `portal.forms.validation.houseNr`

File: alv-portal-ui/src/app/shared/forms/validation.service.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (emina-str) [reply alv-portal-ui/src/app/shared/forms/validation.service.ts]: Changed

File: alv-portal-ui/src/app/shared/forms/validation.service.ts (hunk 3 above)
Comment (emina-str) [reply alv-portal-ui/src/app/shared/forms/validation.service.ts]: Done

File: alv-portal-ui/src/app/shared/forms/validation.service.ts (hunk 2 above)
Comment (emina-str) [reply alv-portal-ui/src/app/shared/forms/validation.service.ts]: Done

**Pull request merged**